cinema/views.py

```diff
@@ -485,16 +524,36 @@ def get_context_data(self, **kwargs):
         return context
 
 
-class NewsDetailView(DetailView):
-    model = News
+def news_detail(request, pk):
+    news = get_object_or_404(News, pk=pk)
+    context = {
+        'news': news,
+        'imdb_top_5': imdb_top_5,
+        'budget_top_5': budget_top_5,
+        'usa_gross_top_5': usa_gross_top_5,
+        'world_gross_top_5': world_gross_top_5
+    }
+    comments = CommentToNews.objects.filter(news=pk, is_active=True)
```

Comment:
You don't have to do that at all. You have a `news` object, so get comments on the Foreign key right away.

Reply:
Thanks. I did as you recommended to me. Unfortunately, appeared addition query in DB. How i didn't try, but "prefetch_related" didn't cache data, at all(((( BIG PROBLEM FOR ME - prefetch_related